### ISSUE TITLE
[SYCL-MLIR] Investigate `functions.cpp` lit failure with `sycl` commit

### ISSUE THREAD

**Issue by whitneywhtsang** [intel/llvm#10800]
`functions.cpp` fails after e4676b6596fb82ede0995eaeba1f2ce3bf34d576. It is temporarily reverted in https://github.com/intel/llvm/pull/10790. 
```
Failed Tests (1):
  cgeist :: Verification/sycl/functions.cpp
```
```
error: 'sycl.accessor.get_pointer' op result #0 must be memref of any type values, but got '!sycl.multi_ptr<[i32, global, legacy], (memref<?xi32, 1>)>'
```

To reproduce:
```
git revert 3393304b036c10d60fbc2b7a13be47ec61bc93b2
ninja check-cgeist
```

**Comment by victor-eds** [comment]
Thanks, @whitneywhtsang, looking into it

**Comment by victor-eds** [comment]
Fixed in #10802. Reported error is not relevant, as that is intended: we're running verification on SYCL ops being generated to avoid breaking compilation due to mismatches between DPC++ headers and SYCL spec.
